app/api/apply/route.ts

```diff
@@ -1,13 +1,67 @@
 import { NextRequest, NextResponse } from "next/server"
 import { google } from "googleapis"
+import { isEmail } from "@/lib/validation"
 
 export const runtime = "nodejs"
 export const dynamic = "force-dynamic"
 export const revalidate = 0
 
+// In-memory per-IP sliding window rate limiter
+const rateLimitStore = new Map<string, number[]>()
+const WINDOW_MS = 60 * 1000 // 1 minute window
+const MAX_REQUESTS = 5
+
+function getClientIp(req: NextRequest): string {
+  const xff = req.headers.get("x-forwarded-for") || ""
+  const ip = xff.split(",")[0]?.trim() || req.headers.get("x-real-ip") || "unknown"
+  return ip
+}
```

Comment:
better to use `req.ip` in Next.js API 
```suggestion
function getClientIp(req: NextRequest): string {
  const ip = (req as any).ip || 
             req.headers.get("x-forwarded-for")?.split(",")[0]?.trim() || 
             req.headers.get("x-real-ip") || 
             "unknown"
  return ip
}
```